regression/cbmc-incr-smt2/bitvector-bitwise-operators/shift_right.c

```diff
@@ -0,0 +1,24 @@
+int main()
+{
+  // In C, whether a right-shift is arithmetic or logical depends on the
+  // original type being shifted. An unsigned value will be shifted to
+  // the right in a logical manner (this assigns `0` to the leftmost bit).
+  // If the type is signed, right shift will assign the sign bit to the
+  // leftmost digit.
+
+  int bitmask_2 = 0x2;
+  int bitmask_3 = 0x3;
+
+  int first;
+  unsigned int second;
+
+  int place;
+  __CPROVER_assume(place >= 1);
+
+  int result_signed = first >> place;
+  unsigned int result_unsigned = second >> place;
+
+  __CPROVER_assert(result_signed && bitmask_2, "This should be satisfiable");
+  __CPROVER_assert(
+    result_unsigned && bitmask_3, "This should also be satisfiable");
```

Comment:
The `bitmask` values are not being used as bit masks because you are using "logical and"  `&&` instead of "bit wise and"  `&`.

Reply:
Yes, you're right 🤦🏻 I will see to it fixed.

Reply:
Looks like the bit masks are gone completely. Therefore this is no longer an issue.